src/main/webapp/script.js

```diff
@@ -161,6 +161,9 @@ function insertCalendar() {
     const calElement = document.getElementById('caldiv');
 
     fetch('/calendar').then((response) => response.text()).then((calSrc) => {
+        if(!calSrc || 0 === calSrc.length || !calSrc.trim()) {
```

Comment:
no need for 0 === calSrc.length, it's already caught by !calSec (empty string is falsy)

Reply:
updated